WordPress/Sniffs/VIP/SuppressFiltersSniff.php

```diff
@@ -0,0 +1,192 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace WordPress\Sniffs\VIP;
+
+use WordPress\AbstractFunctionParameterSniff;
+use PHP_CodeSniffer_Tokens as Tokens;
+
+/**
+ * Checks for suppress_filters=>false being supplied in get_posts(), wp_get_recent_posts() and get_children().
+ *
+ * @link    https://vip.wordpress.com/documentation/vip/code-review-what-we-look-for/
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   0.14.0
+ */
+class SuppressFiltersSniff extends AbstractFunctionParameterSniff {
+
+	/**
+	 * Functions this sniff is looking for.
+	 *
+	 * @since 0.14.0
+	 *
+	 * @var array
+	 */
+	protected $target_functions = array(
+		'get_posts'           => true,
+		'wp_get_recent_posts' => true,
+		'get_children'        => true,
+	);
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @since 0.14.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+
+		if ( false === $this->target_functions[ $matched_content ] ) {
+			return;
+		}
+
+		// Flag to check if suppress_filters is passed or not.
+		$is_used     = false;
+		$array_value = '';
+
+		// Retrieve the value parameter's details.
+		$argument_data = $this->phpcsFile->findNext( Tokens::$emptyTokens, $parameters[1]['start'], ( $parameters[1]['end'] + 1 ), true );
+
+		// When the list of argument were passed through variable.
+		// eg.  $args = array( 'foo' => 'bar' ), $args['foo'] => 'bar'.
+		if ( T_VARIABLE === $this->tokens[ $argument_data ]['code'] ) {
```

Comment:
@jrfnl Yes, this block is not checking value when passing argument as you mentioned, It would be much appreciated if could you please give me any reference/sniff file where this already had been done?